prime-angular-frontend/src/app/modules/enrolment/pages/overview/overview.component.ts

```diff
@@ -196,8 +197,15 @@ export class OverviewComponent extends BaseEnrolmentPage implements OnInit {
    * enrolment for checking validation instead of form state.
    */
   private getEnrolmentErrors(enrolment: Enrolment): ValidationErrors {
+    const isDeviceProvider = this.enrolmentService.enrolment.careSettings.some((careSetting) =>
+      careSetting.careSettingCode === CareSettingEnum.DEVICE_PROVIDER);
+    const hasDeviceProviderIdentifier = this.enrolmentService.enrolment.deviceProviderIdentifier;
+
     return {
-      certificateOrOboSite: !enrolment.certifications?.length && !enrolment.oboSites?.length
+      certificate: !enrolment.certifications?.length,
+      certificateOrOboSite: !enrolment.certifications?.length && !enrolment.oboSites?.length,
+      deviceProvider: isDeviceProvider && !hasDeviceProviderIdentifier,
+      deviceProviderOrOboSite: (isDeviceProvider && !hasDeviceProviderIdentifier) && !enrolment.oboSites?.length
```

Comment:
You've update the enrolment errors to be displayed, but you're not gating the submission with that information.  Recently I added a new method to the AbstractEnrolmentFormState to allow for a specific `isValidSubmission` to be applied.  You'll want to make sure enrolments follows a similar pattern so we can gate submissions, as well as apply the error check to the submission button.

Reply:
Fixed